bentoml/_internal/server/grpc_app.py

```diff
@@ -42,6 +45,37 @@ def __init__(
         self.bento_service = bento_service
         self.enable_metrics = enable_metrics
 
+    @inject
+    async def wait_for_runner_ready(
+        self,
+        *,
+        runner_timeout: int = Provide[BentoMLContainer.runners_config.timeout],
+        check_interval: int = 5,
+    ):
+        start_time = time.time()
+        logger.debug(
+            "Waiting for runners {!r} to be ready...".format(self.bento_service.runners)
+        )
+        while time.time() - start_time < runner_timeout:
+            try:
+                if all(
+                    await asyncio.gather(
+                        *(
+                            runner.runner_handle_is_ready()
+                            for runner in self.bento_service.runners
+                        )
+                    )
+                ):
+                    break
+                else:
+                    time.sleep(check_interval)
+            except (ConnectionError, socket.timeout) as e:
+                logger.debug("[%s] Retrying ..." % e)
+                time.sleep(check_interval)
+        raise RuntimeError(
+            f"Timed out waiting {runner_timeout} seconds for runners to be ready."
```

Comment:
Why are we using runner timeout for the startup timeout? I don't think we should timeout by default, but if we do, the timeout for this and runner requests should definitely be separate fields.

Reply:
I'll let @aarnphm chime in since he authored this and can speak to it a lot better than I can.

@sauyon For my learning, what do you suggest to be the behavior instead of "timeout by default"? And could you clarify "timeout for this and runner requests should definitely be separate fields" for me?

Reply:
I think timeout for this make sense, as if the runners failed to start, we should just raise RuntimeError or at least return a FAILED_PRECONDITION

Reply:
Per our discussion, I removed timeout in a recent commit.